File: CoreService/src/main/java/com/example/demo/model/Market/Product.java
package com.example.demo.model.Market;

import lombok.*;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    private Long id;
    private String productCode;
    private String title;
    private String description;
    private int price;
    private String email; // FK (Users 테이블)
    private Long categoryId; // FK (Categories 테이블)
    private Long hobbyId;  // FK (Hobbies 테이블)
    private String transactionType;
    private String registrationType;
    private int maxParticipants;  // 모집인원
    private int currentParticipants; // 현재 모집된 인원
    private boolean isVisible; // 모집 가능 여부
    private String days; // 추가
    private LocalDateTime startDate;  // 일정 시작일
    private LocalDateTime endDate;  // 일정 종료일
    private Double latitude; // 거래 장소 위도 (대면 거래일 경우 필수)
    private Double longitude; // 거래 장소 경도 (대면 거래일 경우 필수)
    private String meetingPlace; // 거래 장소명 (대면 거래일 경우 필수)
    private String address; // 거래 장소 주소 (선택적)
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // Join 한 유저 정보
    private String nickname;
    private String bio;
    private int dopamine;

    // Join 한 이미지 정보
    private List<String> imagePaths;  // 이미지 리스트
    private String thumbnailPath;  // 대표 이미지
}

File: vite-react-teamsketch/src/types/product.ts
// 상품 등록 요청 타입
export interface IProductRegisterRequest {
  title: string;
  description: string;
  price: number;
  categoryId: number;
  hobbyId: number;
  maxParticipants: number;
  transactionType: '대면' | '비대면';
  registrationType: '구매' | '판매';
  startDate: string;
  endDate: string;
  latitude?: number;
  longitude?: number;
  meetingPlace?: string;
  address?: string;
  days: string[];
  images?: File[];
}

// 상품 응답 타입
export interface IProduct {
  id: number;
  productCode: string;
  title: string;
  description: string;
  price: number;
  email: string;
  categoryId: number;
  hobbyId: number;
  transactionType: '대면' | '비대면';
  registrationType: '구매' | '판매';
  maxParticipants: number;
  currentParticipants: number;
  days: string[];
  startDate: string;
  endDate: string;
  latitude: number | null;
  longitude: number | null;
  meetingPlace: string | null;
  address: string | null;
  createdAt: string | null;
  imagePaths: string[];
  thumbnailPath: string | null;
  nickname: string | null;
  bio: string | null;
  dopamine: number;
  visible: boolean;
}

// API 응답 타입
export interface IApiResponse<T> {
  status: 'success' | 'error';
  message: string;
  data: T;
}

// 상품 등록 응답 타입
export type IProductRegisterResponse = IApiResponse<IProduct>;

// 상품 목록 조회 응답 타입
export type IProductListResponse = IApiResponse<IProduct[]>;
